projects/rulesets-base/src/rule-runner/assertions.ts

```diff
@@ -77,13 +77,19 @@ class AssertionRunner<T extends AssertionType> implements Assertions<T> {
       | 'addedAssertions'
       | 'changedAssertions'
       | 'removedAssertions'
+      | 'addedAndChangedAssertions'
   ): AssertionTypeToHelpers[T] => {
     const registerAssertion = (...args) => {
       const assertion = args[1] || args[0];
       if (assertionKey === 'changedAssertions') {
         this.changedAssertions.push((before, after) => {
           assertion(after);
         });
+      } else if (assertionKey === 'addedAndChangedAssertions') {
+        this.addedAssertions.push(assertion);
+        this.changedAssertions.push((before, after) => {
+          assertion(after);
```

Comment:
@niclim  -- this is the part that's really most consequential right? `addedOrChanged` rules always get the `after`? 

That seems like the most useful behavior to me. You could write the rule to assert the final state of something in the spec. 

The one *weird thing that may happen is that the rule triggers when you change something completely unrelated. Like I change description and now it's enforcing the name of an existing property. That can happen with changed rules too, but you can be smarter about it since you have both. 

I don't think it's a problem that should block this, but we should keep our ears open

Reply:
Yes, you care about the final state of the spec. I think if you did want to enforce more specific change assertions, you would split out added and changed into separate assertions.

This is really a shorthand of what you would be doing for added or changed.